docs/lambda-selector/extension-arn-replacement.asciidoc

```diff
@@ -2,6 +2,8 @@
 <script>
 window.addEventListener("DOMContentLoaded", async () => {
   addArnGenerator('extension', 'apm-aws-lambda', 'arn:aws:lambda:${region}:267093732750:layer:elastic-apm-extension-${version}-${arch}');
+  replaceAgentDockerImageParams('FROM docker.elastic.co/observability/apm-agent-nodejs:latest AS nodejs-agent',
+                                'COPY --from=nodejs-agent /opt/nodejs/ /opt/nodejs/');
```

Comment:
I don't think this is the right fix. 

Support for calling `replaceAgentDockerImageParams` was added in https://github.com/elastic/apm-aws-lambda/pull/204 (by @AlexanderWert) and I believe the intent there was for the docs in the *agent* repos to call `replaceAgentDockerImageParams` as appropriate. For example:

1. Here is where the apm-agent-nodejs docs are attempting to call it: https://github.com/elastic/apm-agent-nodejs/pull/2747/files#diff-6806c28790a7d5d35109c7c1e989489930f8a57ff89a6329c60092398bc04863R5-R6
2. Here is where the apm-agent-java docs are calling it: https://github.com/elastic/apm-agent-java/pull/2658/files#diff-72367a196c9de5ca6cadc18c850ea6d4fabe5d0510eed2f10788572c22290d48R5-R6

It looks like this is working for the *Java* agent docs: https://www.elastic.co/guide/en/apm/agent/java/current/aws-lambda.html#_step_2_add_the_apm_layers_to_your_lambda_function

But not for the Node.js agent docs: https://www.elastic.co/guide/en/apm/agent/nodejs/current/lambda.html#_step_2_add_the_apm_layers_to_your_lambda_function

So, the question is why this isn't working for the Node.js APM agent docs.

Reply:
> So, the question is why this isn't working for the Node.js APM agent docs.

When I view the browser console when clicking the "Container Image" tab on https://www.elastic.co/guide/en/apm/agent/nodejs/current/lambda.html, I see this exception:
![Screen Shot 2022-07-14 at 8 51 48 AM](https://user-images.githubusercontent.com/46866/179024384-6db4072e-e7da-47af-af48-f1c7f30e8213.png)

Reply:
That stacktrace is 50k into an inscrutable bundle, and I guess the doc pages don't serve sourcemap files for debugging (boo).

I can manually use that `replaceAgentDockerImageParams` function in the console in the exact way the apm-agent-nodejs docs JS was doing to get the expected result:

![Screen Shot 2022-07-14 at 9 00 18 AM](https://user-images.githubusercontent.com/46866/179026259-340f2a52-80da-433b-a6f0-681dd5973d29.png)

My guess is there is a bug in the docs.git repo that is being tickled with the usage in the apm-agent-nodejs docs. /cc @bmorelli25 

Looking at the minified bundle code for the function `Od` in which the exception was thrown:

```js
var Od=function(e){var t=e.parentNode,r=t.parentNode;t.querySelectorAll("[aria-selected=\"true\"]"). // ...
```

perhaps this could be one of:
- this `setActiveTab` function: https://github.com/elastic/docs/blob/0b1e761e2f9424f7acaad147887d888b0751ed80/resources/web/docs_js/components/tabbed_widget.js#L36-L53, or
- the `setActiveTab` function in this asciidoc file https://raw.githubusercontent.com/elastic/docs/master/shared/cloud/tab-widgets/code.asciidoc

Reply:
Why "master" (i.e. the state of the "main" branch) has this and "current" (a.k.a., "3.x") does not:

I think this is because I missed part of the process for the v3.36.0 release of the agent. The last step of the [release process](https://github.com/elastic/apm-agent-nodejs/blob/main/CONTRIBUTING.md#current-major) for the Node.js APM agent is to reset the "3.x" branch to the tagged commit. It appears that I forgot to do that as part of the [3.36.0 release](https://github.com/elastic/apm-agent-nodejs/pull/2776). 

`git log ...` shows:

```
* f5307cd5 - (HEAD -> main, origin/main) docs: consistent naming (#2815) (7 days ago)
...
* 7d3754a1 - (tag: v3.36.0) 3.36.0 (#2776) (4 weeks ago)
...
* 7b2dddf5 - Added container image tab to lambda setup docs (#2747) (4 weeks ago)
...
* 9ba49222 - (tag: v3.35.0, origin/3.x) fix: the "Push Docker Image" jenkins step requires a HOME var (#2748) (6 weeks ago)
```

^^^ The `origin/3.x` branch is pointed to the v3.35.0 tag, instead of to the v3.36.0 tag. As a result, it is missing commit 7b2dddf5 that added the agent part of the lambda docs thing. My bad. I'll fix that now.

Reply:
> I'll fix that now.

```
% git branch -f 3.x v3.36.0

% git push origin 3.x
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
To github.com:elastic/apm-agent-nodejs.git
   9ba49222..7d3754a1  3.x -> 3.x
```

I'm not sure what CI may/will update the "current" docs for the APM Node.js agent.

Reply:
> I'm not sure what CI may/will update the "current" docs for the APM Node.js agent.

This has been rebuilt now, as is, I believe shown by these lines from https://elasticsearch-ci.elastic.co/view/Docs/job/elastic+docs+master+build/25385/console

```
11:09:26 INFO:build_docs:Building docs
11:09:38 INFO:build_docs: -                        APM Node.js Agent: Building 3.x...
11:09:39 INFO:build_docs: -                        APM Node.js Agent: Finished 3.x
```

https://www.elastic.co/guide/en/apm/agent/nodejs/current/lambda.html now has a "Container Image" tab in both tab-groups.  So this PR can be closed.

@bmorelli25 Do you think it is worth having a separate issue for having https://github.com/elastic/docs/blob/0b1e761e2f9424f7acaad147887d888b0751ed80/resources/web/docs_js/components/tabbed_widget.js#L55-L85 not throw if a particular tab-group doesn't have a tab with the given index? I don't have a strong opinion.

Reply:
Great detective work 🔍.

> Do you think it is worth having a separate issue for having https://github.com/elastic/docs/blob/0b1e761e2f9424f7acaad147887d888b0751ed80/resources/web/docs_js/components/tabbed_widget.js#L55-L85 not throw if a particular tab-group doesn't have a tab with the given index? I don't have a strong opinion.

My honest answer is that I don't know if this is worth changing. I'll follow up with docs infra and open an issue based on that conversation.